service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/EhrResendController.java

```diff
@@ -0,0 +1,136 @@
+import org.hl7.fhir.dstu3.model.CodeableConcept;
+import org.hl7.fhir.dstu3.model.Coding;
+import org.hl7.fhir.dstu3.model.Meta;
+import org.hl7.fhir.dstu3.model.OperationOutcome;
+import org.hl7.fhir.dstu3.model.UriType;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import uk.nhs.adaptors.gp2gp.common.service.FhirParseService;
+import uk.nhs.adaptors.gp2gp.common.service.RandomIdGeneratorService;
+import uk.nhs.adaptors.gp2gp.common.service.TimestampService;
+import uk.nhs.adaptors.gp2gp.common.task.TaskDispatcher;
+import uk.nhs.adaptors.gp2gp.ehr.model.EhrExtractStatus;
+import uk.nhs.adaptors.gp2gp.gpc.GetGpcStructuredTaskDefinition;
+
+import java.util.Collections;
+import java.util.List;
+
+@Slf4j
+@RestController
+@AllArgsConstructor(onConstructor = @__(@Autowired))
+@RequestMapping(path = "/ehr-resend")
+public class EhrResendController {
+
+    private static final String OPERATION_OUTCOME_URL = "https://fhir.nhs.uk/STU3/StructureDefinition/GPConnect-OperationOutcome-1";
+    public static final String INTERNAL_SERVER_ERROR = "INTERNAL_SERVER_ERROR";
+    public static final String INVALID_IDENTIFIER_VALUE = "INVALID_IDENTIFIER_VALUE";
+
+    private EhrExtractStatusRepository ehrExtractStatusRepository;
+    private TaskDispatcher taskDispatcher;
+    private RandomIdGeneratorService randomIdGeneratorService;
+    private final TimestampService timestampService;
+    private final FhirParseService fhirParseService;
+
+    @PostMapping("/{conversationId}")
+    public ResponseEntity<String> scheduleEhrExtractResend(@PathVariable String conversationId) {
+        EhrExtractStatus ehrExtractStatus = ehrExtractStatusRepository.findByConversationId(conversationId).orElseGet(() -> null);
+
+        if (ehrExtractStatus == null) {
+            var details = getCodeableConcept(INVALID_IDENTIFIER_VALUE);
+            var diagnostics = "Provide a conversationId that exists and retry the operation";
+
+            var operationOutcome = createOperationOutcome(OperationOutcome.IssueType.VALUE,
+                                                          OperationOutcome.IssueSeverity.ERROR,
+                                                          details,
+                                                          diagnostics);
+            var errorBody = fhirParseService.encodeToJson(operationOutcome);
+
+            return new ResponseEntity<>(errorBody, HttpStatus.NOT_FOUND);
+        }
+
+        if (noErrorsInEhrReceivedAcknowledgement(ehrExtractStatus) && ehrExtractStatus.getError() == null) {
+
+            var details = getCodeableConcept(INTERNAL_SERVER_ERROR);
```

Comment:
Pretty sure we should be using returning a [202: Accepted](https://http.cat/status/202) here and not a 500:

> The HTTP 202 Accepted response status code indicates that the request has been accepted for processing, but the processing has not been completed; in fact, processing may not have started yet.

Reply:
This situation doesn't look like an [INTERNAL_SERVER_ERROR](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/500).

> The HTTP 500 Internal Server Error [server error response](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#server_error_responses) status code indicates that the server encountered an unexpected condition that prevented it from fulfilling the request. This error is a generic "catch-all" response to server issues, indicating that the server cannot find a more appropriate [5XX error](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#server_error_responses) to respond with.
> 
> If you're a visitor seeing 500 errors on a web page, these issues require investigation by server owners or administrators. There are many possible causes of 500 errors, including: improper server configuration, out-of-memory (OOM) issues, unhandled exceptions, improper file permissions, or other complex factors. Server administrators may proactively log occurrences of server error responses, like the 500 status code, with details about the initiating requests to improve the stability of a service in the future.

A 400 style error would be a lot more appropriate. This is a client request which is invalid given the current DB state.